codex-rs/tui/src/lib.rs

```diff
@@ -142,22 +142,39 @@ pub fn run_main(
         .with(tui_layer)
         .try_init();
 
-    let show_login_screen = should_show_login_screen(&config);
+    let show_login_screen = should_show_login_screen(&config).await;
+    if show_login_screen {
+        std::io::stdout().write_all(
+            b"Oh dear, we don't seem to have an API key.\nTerribly sorry, but may I open a browser window for you to log in? [Yn] ",
+        )?;
+        std::io::stdout().flush()?;
+        let mut input = String::new();
+        std::io::stdin().read_line(&mut input)?;
+        let trimmed = input.trim();
+        if !(trimmed.is_empty() || trimmed.eq_ignore_ascii_case("y")) {
```

Comment:
My mind wants to deMorgan this to...

```suggestion
        if !trimmed.is_empty() && !trimmed.eq_ignore_ascii_case("y") {
```

Reply:
hm, i prefer the un-deMorgan'd version. `if !(is_yes)` vs `if !is_yes_v1 || !is_yes_v2` seems simpler.